documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md

```diff
@@ -0,0 +1,202 @@
+      `domain.spec.configuration.model.onlineUpdate.onNonDynamicChanges` is `CommitUpdateOnly`.
+    * The domain resource attribute
+      `domain.spec.configuration.model.onlineUpdate.enabled` is `True`.
+    * There were model changes and these changes modify non-dynamic WebLogic configuration.
+    * Processing successfully completed, including the introspector job.
+    * The administrator has not subsequently rolled/restarted each WebLogic Server pod
+      (to propagate the pending non-dynamic changes).
+  * See [Online update status and labels]({{< relref "managing-domains/model-in-image/runtime-updates#online-update-status-and-labels" >}})
+    on how to see which pods are awaiting restart using WebLogic pod labels.
+{{%expand "Click here for an example of a domain status with a ConfigChangesPendingRestart condition." %}}
+```
+Status:
+  ...
+  Conditions:
+    Last Transition Time:  2021-01-20T15:09:15.209Z
+    Message:               Online update completed successfully, but the changes require restart and the domain resource specified 'spec.configuration.model.onlineUpdate.onNonDynamicChanges=CommitUpdateOnly' or not set. The changes are committed but the domain require manually restart to  make the changes effective. The changes are: Server re-start is REQUIRED for the set of changes in progress.
+    
+    The following non-dynamic attribute(s) have been changed on MBeans
+    that require server re-start:
+    MBean Changed : com.bea:Name=oracle.jdbc.fanEnabled,Type=weblogic.j2ee.descriptor.wl.JDBCPropertyBean,Parent=[sample-domain1]/JDBCSystemResources[Bubba-DS],Path=JDBCResource[Bubba-DS]/JDBCDriverParams/Properties/Properties[oracle.jdbc.fanEnabled]
+    Attributes changed : Value
+    Reason:                      Online update applied, introspectVersion updated to 82
+    Status:                      True
+    Type:                        ConfigChangesPendingRestart
+```
+{{% /expand %}}
+
+- `Rolling`
+  * This condition indicates that the operator is rolling the server pods in a domain, such as after it
+    has detected an update to the domain resource or Model in Image model that require it to
+    perform a [rolling restart]({{< relref "managing-domains/domain-lifecycle/startup#rolling-restarts" >}}) of the domain.
+  * The `status` attribute is always `True` for a `Rolling` condition.
+  * The `Rolling` condition is removed from the domain status when the rolling is completed.
+
+
+### Types of cluster conditions
+
+The following is a list of condition types for a cluster resource.
+- `Completed`
+    * The `status` attribute of a `Completed` condition indicates whether the desired state of the
+      cluster resource has been fully achieved.
+    * The `status` attribute is set to `True` when:
+        * All server pods in the cluster that are expected to be running are ready at their target,
+          images(s), `restartVersion`, and `introspectVersion`
+        * There are no pending server shutdown requests.
```

Comment:
It'd be helpful, in my opinion, to document the lifecycle of all conditions. Here's the background:

- Once the operator detects a Cluster or Domain resource, it will ensure that each resource always has exactly one occurrence of an `Available` and of a `Completed` condition. The operator will not subsequently remove these conditions but instead will change their `status` to `True` or `False` as appropriate. You can use this fact as a way to detect when an operator has detected a Domain or Cluster resource.

- The `Failure`, `Rolling`, and `ConfigChangesPendingRestart` conditions are ephemeral: they are only added when they apply, and are removed when they no longer apply. 

- The `Failure` condition is the only condition that can have multiple occurrences. This occurs when there are multiple concurrent failures of different types.

You might want to roll the 'generation' information that I mentioned elsewhere into this same discussion.

Reply:
You might also want to mention in this same location that changes to `Available`, `Completed`, `Failure`, and `Rolling` have corresponding events (with link).